### ISSUE TITLE
Figure out how Thimble should respond to different types of URLs

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.webmaker.org#1120]
Placeholder issue - will fill out after my meeting.

**Comment by flukeout** [comment]
A related issue in #1130 - when you try to load a deleted or otherwise not available project in the editor.

**Comment by humphd** [comment]
Good example of confusion around how to share something a user makes here:

https://twitter.com/smithisgeneric/status/644924701006004224

This editor URL looks like it will do something, but it won't.  How to convey that it's like a localhost url?

**Comment by humphd** [comment]
I'm not sure where to put this, but this HN post has come great comments about ways to mix human-readable URLs with numeric IDs: https://news.ycombinator.com/item?id=10244129  Basically, you have your server ignore everything after the id part you care about, which lets users create much more readable URLs, but lets you have unique IDs.  We should consider this.

**Comment by humphd** [comment]
I saw more of this on twitter recently, especially people sharing the `/anonymous/*` links.  I wonder if we should change the word `anonymous` to something that better indicates "this isn't available to anyone but you on this computer".

**Comment by hannahkane** [comment]
Are the options @flukeout mentioned in the original comment possible? I like a combination of the first and third bullet points.

**Comment by gideonthomas** [comment]
https://github.com/mozilla/thimble.webmaker.org/issues/1259 has some more discussion points

**Comment by humphd** [comment]
Another example of user confusion: https://twitter.com/fth_nix/status/684468236004540416

**Comment by flukeout** [comment]
Could we take a similar approach for this as we've decided on in cases where the published project doesn't have an index.html page?

* https://github.com/mozilla/thimble.mozilla.org/issues/1569

If we detect someone trying to open a URL to the editor (and they're not signed in, or they're a different user) we can show a custom error page instead?

If so, I can design that page and this might make a good first bug.

cc @gideonthomas @humphd

**Comment by gideonthomas** [comment]
Yeah I think that makes sense!

**Comment by XuanWang-Amos** [comment]
Hi all, I'd like to work on this issue, since I fell it will be a good start for me to get acquainted with thimble, is that OK?

**Comment by humphd** [comment]
@XuanWang-Amos indeed, please do give it a try!  Welcome to Thimble.  Ask us questions as you get started and need help, here or in https://chat.mozillafoundation.org/mozilla/channels/thimble.

**Comment by XuanWang-Amos** [comment]
@humphd , hi humphd, I already joined the channel, and I have some questions regarding this issue.
1. I can think of 4 situations:
    1.1 user open other users project.
    1.2 user open anonymous project.
    1.3  anonymous open other users project.
    1.4  anonymous open anonymous project.
1.2, 1.3 and 1.4 are already deals with, 1.2 the project will be upgraded, 1.3 and 1.4 will show them thimble, so should the user be redirected in those situations too or just redirect situation 1.1?

2. what page should the user be redirected to?

**Comment by XuanWang-Amos** [comment]
@humphd , hi humphd, I already joined the channel, and I have some questions regarding this issue.
1. right now, if a login user open an anonymous project,  the project will be upgraded, should we keep it this way?
2. what page should the user be redirected to?

**Comment by flukeout** [comment]
Hi @XuanWang-Amos, thanks for taking this on, let's start with the following. If someone tries to load up the 'editor' version of someone's project (ex.. ``https://thimble.mozilla.org/en-US/user/flukeout/841612``) - let's do this...

* If the associated project is published, we can redirect the user to the published version
* If it's unpublished, we can say "OOPS, this is someone's draft!"

**Comment by XuanWang-Amos** [comment]
@flukeout , sure, I can start with that.  after looking through the code, I'm thinking we can redirect by project's 'publish_url' property, but the project data is sent from publish.wm.org, if the user who open the url is not the owner, the server will retrun "401 Unauthorized", is there any other way to get the publish_url?
and also, if the project is published but the user made some changes and not re-publish yet, open publish_url will not include the latest change, this might be a problem.

**Comment by flukeout** [comment]
@gideonthomas is there any way to get the publish_url of a project that isn't yours?

If we are able to get this URL, I think it's OK that the project might be out of date. I still think it's better than not trying to show it.

**Comment by gideonthomas** [comment]
@XuanWang-Amos looks like we need to implement something on the `publish.wm.org` repo - a route that allows you to get a published project by its project ID vs. it's published project ID. Once we implement that route, you should hit that route (it'll probably be something like `/publishedProjects?project_id=xxx`), you won't need to be authenticated for that. You need to check if the response status code is `HTTP 404`. If it is `HTTP 404`, you need to return a page that says "OOPS, this is someone's draft!" (@flukeout can advise on the design). If you actually get a response that is `HTTP 200`, you will have access to the publish_url in the response body and you redirect the user to that url.

My suggestion to get you started, until that route has been implemented in the publish.wm.org repo,  just write something that simulates it:
```js
var fakeResponseStatusCode = projectId % 2 === 0 ? 200 : 404; // we're faking the response status code based on whether the project id was even or not.
var fakeResponseBody = { publish_url: "https://fakeurl.com" };

// rest of your logic based on the response here
```
This way, you can start designing the "OOPS, this is someones draft" page and add the rest of the logic without being blocked on something else being implemented.

**Comment by XuanWang-Amos** [comment]
@gideonthomas sure, I'll start working on that, @flukeout can you give me some advice on the page design? may be it will look like the page missing `index.html`?

**Comment by humphd** [comment]
@XuanWang-Amos it looks like this work got paused.  I'm going to move this bug to an unassigned state, and if someone (including you) wants to pick it up in the future, we'll re-assign.
